test/snapd_test.dart

```diff
@@ -793,6 +796,43 @@ class MockSnapdServer {
     _writeSyncResponse(request.response, snaps);
   }
 
+  void _processFindById(HttpRequest request) async {
```

Comment:
The tests try not to have hard-coded behaviour like this - instead make a MockAssertionObject that is provided in the MockSnapdConstructor (in a map or similar). Then the contents of the assertion will be in the test.